### DISCUSSION TITLE
UserWhereUniqueInput with case insensitive ?

### DISCUSSION THREAD

**Discussion by ddavid67** [prisma/prisma#5339] [Q&A] [answered]
Hello,

I am using upsert / connectOrCreate functions to quickly create some tags record if necessary. (awesome feature by the way)
I am using postgresql so everything is case sensitive.

I want to be able to have tags like "tagname" or "SUPERtagname" (keeping casing) but I don't want to be able to have duplicate record with same name like "TAGNAME" or "supertagname". 
The first tag to be insert should be the "true one".

Problem is the when using  upsert / connectOrCreate, these functions are using the UserWhereUniqueInput type, so I am not able to use insensitive mode with equals on this where selection

Any suggestions at how to achieve this ? 
If a tag already exist, I want to connect/update the first one.

**Comment by ryands17** [comment, marked as answer]
@daeschlerdavid 👋 

One way to do this would be to have two fields in your model. One for the actual value and the other with the same value but lowercase. The lowercase field would be unique and so when using `connectOrCreate`, you can directly pass that lowercase value in as the tag which will be recognized.

Another option is if you're just using a single column is to find the value first and then only connect it if it's found.

**Reply by ddavid67** [reply]
I like the first option.

Maybe a postgresql generated stored column that is LOWER(tag) would make this easier?
Not sure if this kind of column can be unique, and how prisma would manage this.

In any case this seems trivial to implement in application side

Thanks you !

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
